cli/run/index.ts

```diff
@@ -118,7 +118,68 @@ async function execute(
 			if (command.stdin !== false) {
 				inputOptions.plugins!.push(stdinPlugin());
 			}
+			if (command.plugin) {
+				const plugins = Array.isArray(command.plugin) ? command.plugin : [command.plugin];
+				for (const plugin of plugins) {
+					if (/[={}]/.test(plugin)) {
+						// -p plugin=value
+						// -p "{transform(c,i){...}}"
+						loadAndRegisterPlugin(inputOptions, plugin)
+					} else {
+						// split out plugins joined by commas
+						// -p node-resolve,commonjs,buble
+						plugin.split(",").forEach((plugin: string) => loadAndRegisterPlugin(inputOptions, plugin));
+					}
+				}
+			}
 			await build(inputOptions, outputOptions, warnings, command.silent);
 		}
 	}
+
+	function loadAndRegisterPlugin(inputOptions: InputOptions, pluginText: string) {
+		let plugin : any = null;
+		let pluginArg : any = undefined;
+		if (pluginText[0] === '{') {
+			// -p "{transform(c,i){...}}"
+			plugin = new Function('return ' + pluginText);
+		} else {
+			const match = pluginText.match(/^([@.\/\\\w|^{}|-]+)(=(.*))?$/);
+			if (match) {
+				// -p plugin
+				// -p plugin=arg
+				pluginText = match[1];
+				pluginArg = new Function('return ' + match[3])();
+			} else {
+				throw new Error(`Invalid --plugin argument format: ${JSON.stringify(pluginText)}`);
+			}
+			if (!/^\.|[@\/\\]/.test(pluginText)) {
```

Comment:
Not sure I completely understand this regexp. Why would I want to try adding prefixes to a path that starts with a `.` but contains `@` or slashes? Why not just `if (!pluginText.startsWith('.'))...`?

Reply:
I think you've missed the or. Only trying the plugin prefixes if it does not start with a dot, or if it does not contain an @ or slashes - only node resolve should be used in those cases.

Reply:
Upon re-reading your comments, you may have missed the leading `!` in the condition. Although this `if` is not strictly needed, it is an optimization to avoid expensive file system checks in cases where a prefix is not possible such as `"foo/bar"` or `"../baz"`. Relative paths and any path with a slash or `@` will never be a prefix candidate.

Reply:
Oh, it is even worse, I saw the `!` but forgot how basic logic works when I negate an OR, don't ask 🙄